app/controllers/returns/ImportedPlasticPackagingController.scala

```diff
@@ -35,48 +36,51 @@ import scala.concurrent.{ExecutionContext, Future}
 class ImportedPlasticPackagingController @Inject() (
   override val messagesApi: MessagesApi,
   cacheConnector: CacheConnector,
-  navigator: Navigator,
   identify: IdentifierAction,
   getData: DataRetrievalAction,
   requireData: DataRequiredAction,
-  form: ImportedPlasticPackagingFormProvider,
+  formProvider: ImportedPlasticPackagingFormProvider,
   val controllerComponents: MessagesControllerComponents,
-  view: ImportedPlasticPackagingView
+  view: ImportedPlasticPackagingView,
+  returnsNavigator: ReturnsJourneyNavigator
 )(implicit ec: ExecutionContext)
     extends FrontendBaseController with I18nSupport {
 
   def onPageLoad(mode: Mode): Action[AnyContent] =
-    (identify andThen getData andThen requireData).async {
+    (identify andThen getData andThen requireData) {
       implicit request =>
-        val preparedForm = request.userAnswers.get(ImportedPlasticPackagingPage) match {
-          case None        => form()
-          case Some(value) => form().fill(value)
-        }
+        val preparedForm = request.userAnswers.fill(ImportedPlasticPackagingPage, formProvider())
+        
         request.userAnswers.get[TaxReturnObligation](ObligationCacheable) match {
-          case Some(obligation) => Future.successful(Ok(view(preparedForm, mode, obligation)))
-          case None             => Future.successful(Redirect(controllers.routes.IndexController.onPageLoad))
+          case Some(obligation) => Ok(view(preparedForm, mode, obligation))
```

Comment:
do we need obligation here? It seem that the view does not uses.

Reply:
Oh... haven't looked, but will have a  👀 thanks

I think we still require that there is an ob in the cache -> that's how we test if a user has correctly started a return vs url-hopped

Reply:
You could use fold as It will still get an obligation, but you would not need to pass it to the view so it could be removed from the view as it is not use in the view (this will make test easier too - less setting to think about)

in onPageLoad:
request.userAnswers.get[TaxReturnObligation](ObligationCacheable).fold(
          Future.successful(Redirect(controllers.routes.IndexController.onPageLoad)))(_ => Future.successful(Ok(view(preparedForm, mode))))

in onSubmit:
 if(request.userAnswers.get[TaxReturnObligation](ObligationCacheable).isEmpty) {
          throw new IllegalStateException("Must have an obligation to Submit against")
        }

or could still use a fold here too.

Just a thought